### PR TITLE
Issue #549 - Add a LifeCycledAdapter to simplify writing LifeCycled anonymous-class

### PR DESCRIPTION
**Pull request by mathieucarbou**
Linked to issue #549

### CODE REVIEW

File: api/src/main/java/org/ehcache/spi/LifeCycledAdapter.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.spi;
+
+/**
+ * @author Mathieu Carbou
+ */
+public class LifeCycledAdapter implements LifeCycled {
```
Comment (ljacomet) [inline api/src/main/java/org/ehcache/spi/LifeCycledAdapter.java]: Make it abstract

**Comment by mathieucarbou** [comment]
@ljacomet : done.

File: api/src/main/java/org/ehcache/spi/LifeCycledAdapter.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.spi;
+
+/**
+ * @author Mathieu Carbou
+ */
+public abstract class LifeCycledAdapter implements LifeCycled {
```
Comment (ljacomet) [inline api/src/main/java/org/ehcache/spi/LifeCycledAdapter.java]: Missed it was in the API module.
I believe we do not want that on the basis of not having concrete classes in the API.
core would be a better choice.

@chrisdennis @alexsnaps @lorban @AbfrmBlr @albinsuresh @rishabhmonga opinions?

**Comment by mathieucarbou** [comment]
@ljacomet : added javadoc. 
NB: The build failed because I amended my last commit and forced push.

**Comment by ljacomet** [comment]
test this please

File: api/src/main/java/org/ehcache/spi/LifeCycledAdapter.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ */
+
+package org.ehcache.spi;
+
+/**
+ * Adapter class for {@link LifeCycled} in case you do not need to implement all methods from the interface.
+ *
+ * @author Mathieu Carbou
+ */
+public abstract class LifeCycledAdapter implements LifeCycled {
```
Comment (ljacomet) [inline api/src/main/java/org/ehcache/spi/LifeCycledAdapter.java]: As said, this should be in `core` module, no `api`

**Comment by ljacomet** [comment]
test this please

**Comment by mathieucarbou** [comment]
@ljacomet : moved to core and added test :-) 
nb: I don't really know what to test for an adapter class... So I tested possible implementations.

**Comment by ljacomet** [comment]
@mathieucarbou Sorry, the sentence is used to trigger a rebuild, not addressed at you :smiley:

**Comment by mathieucarbou** [comment]
test this please

**Comment by mathieucarbou** [comment]
test this please

**Pull request merged**